src/core/openthread.cpp

```diff
@@ -112,6 +112,7 @@ const uint8_t *otGetExtendedPanId(void)
 void otSetExtendedPanId(const uint8_t *aExtendedPanId)
 {
     sThreadNetif->GetMac().SetExtendedPanId(aExtendedPanId);
+    sThreadNetif->GetMle().SetMeshLocalPrefix(aExtendedPanId);
```

Comment:
Is this change related to the scans? If not, it might make sense to split this change out into a separate commit.